server.go

```diff
@@ -2104,10 +2104,22 @@ func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User, ch
 		var i int
 		var stmt influxql.Statement
 		for i, stmt = range q.Statements {
-			// Set default database and policy on the statement.
-			if err := s.NormalizeStatement(stmt, database); err != nil {
-				results <- &Result{Err: err}
-				break
+			// If a default database wasn't passed in by the caller,
+			// try to get it from the statement.
+			defaultDB := database
+			if defaultDB == "" {
+				if s, ok := stmt.(influxql.HasDefaultDatabase); ok {
+					defaultDB = s.DefaultDatabase()
```

Comment:
That answers my question. I just wanted to be sure the error cases (if any) had been considered.